server/src/apub/post.rs

```diff
@@ -264,6 +264,92 @@ impl ApubObjectType for Post {
     )?;
     Ok(())
   }
+
+  fn send_remove(&self, mod_: &User_, conn: &PgConnection) -> Result<(), Error> {
+    let page = self.to_apub(conn)?;
+    let community = Community::read(conn, self.community_id)?;
+    let id = format!("{}/remove/{}", self.ap_id, uuid::Uuid::new_v4());
+    let mut remove = Remove::default();
+
+    populate_object_props(
+      &mut remove.object_props,
+      &community.get_followers_url(),
+      &id,
+    )?;
+
+    remove
+      .remove_props
```

Comment:
@asonix One reason why I'm lazily copying and pasting a lot, is because I don't know a way to access the non-object props in a generic way.

In many cases here it would be useful to create default trait functions using the generic, higher level types, if I could only access their properties. Is there any way to do this?